src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -296,91 +286,141 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
 
     // Validate the legacy names
     // If no legacy name exists use the passed in test name
-    std::string test_name = py;
-    if (test_map.find(py) != test_map.end())
-      test_name = test_map.find(py)->second;
+    std::string test_name = file;
+    if (test_map.find(file) != test_map.end())
+      test_name = test_map.find(file)->second;
 
     // Parse if the file exists here
-    std::string  xrtTestCasePath = "/opt/xilinx/xrt/test/" + test_name;
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
+    xrtTestCasePath = "/opt/xilinx/xrt/test/" + test_name;
+  } else {
+    xrtTestCasePath = "/opt/xilinx/xrt/test/" + file;
+  }
+  boost::filesystem::path xrt_path(xrtTestCasePath);
+  if (!boost::filesystem::exists(xrt_path)) {
+    logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
+    logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
+    _ptTest.put("status", test_token_failed);
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+  return xrtTestCasePath;
+}
 
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+/*
+ * Spawn a testcase process.
+ */
+bool
+spawn_testcase(boost::property_tree::ptree& _ptTest, const std::string& xrtTestCasePath,
+               std::ostringstream & os_stdout, std::ostringstream & os_stderr,
+               std::vector<std::string> args, const test_mode& mode)
+{
+  int exit_code;
+  try {
+    switch (mode) {
+      case test_mode::json:
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      case test_mode::support:
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Checking Test Support", "Test Support Check Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      case test_mode::no_json:
+        if (_ptTest.get<std::string>("file").find(".exe") != std::string::npos)
+          exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        else
+          exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      default:
+        return false; //Mode was not specified correctly.
```

Comment:
```suggestion
        throw std::runtime_exception("Invalid test mode specified"); //Mode was not specified correctly.
```
That way the message will be logged and we can easily find the issue!